React InstantSearch Hooks/query-suggestions/src/Autocomplete.tsx

```diff
@@ -0,0 +1,76 @@
+import { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+import {
+  useInstantSearch,
+  useMenu,
+  useSearchBox,
+} from 'react-instantsearch-hooks-web';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createQuerySuggestionsPlugin } from '@algolia/autocomplete-plugin-query-suggestions';
+
+import '@algolia/autocomplete-theme-classic';
+
+import type { SearchClient } from 'algoliasearch/lite';
+
+type AutocompleteProps = {
+  searchClient: SearchClient;
+};
+
+export function Autocomplete({ searchClient }: AutocompleteProps) {
+  const container = useRef<HTMLDivElement>(null);
+
+  const { setIndexUiState } = useInstantSearch();
+  const { query } = useSearchBox();
+  useMenu({ attribute: 'categories' });
+
+  useEffect(() => {
+    if (!container.current) {
+      return;
+    }
+
+    const querySuggestionsPlugin = createQuerySuggestionsPlugin({
+      indexName: 'instant_search_demo_query_suggestions',
+      searchClient,
+      categoryAttribute: [
+        'instant_search',
+        'facets',
+        'exact_matches',
+        'categories',
+      ],
+      transformSource({ source }) {
+        return {
+          ...source,
+          onSelect({ item }) {
+            setIndexUiState((indexUiState) => ({
+              ...indexUiState,
+              query: item.query,
+              menu: {
+                categories: item.__autocomplete_qsCategory || '',
+              },
+            }));
```

Comment:
I would probably forward Autocomplete props to the component instead of doing it all inside the component, to keep concerns separated.

This way you don't use Hooks directly inside `<Autocomplete>` or do any specific logic. If the user wants to reuse the component differently somewhere else, they can without duplicating it.

Reply:
I fear it would make the sample a bit more complicated.

Since we can't use Hooks on the same component that renders `<InstantSearch>` (App.tsx here), it means we would need to create another layer just to decouple logic from the Autocomplete component.

Do you think we should go ahead with that nevertheless?

Reply:
Yup that's a trade-off, up to you if you think this adds too much complexity.

If we go this route, I think you also don't need to pass the search client as a prop. You can initialize it separately and import it, that should further reduce the noise in app code.

Reply:
I've integrated your proposal in 70983fe